ARCs/arc-0004.md

```diff
@@ -221,23 +243,40 @@ For example:
 A Contract is the complete set of methods that an app implements. It
 is similar to an interface, but may include further details about the
 concrete implementation. All methods in a Contract must be unique
-(specifically, they must have unique method selectors). Method names
-in Contracts **MAY** begin with underscore, but these names are
-reserved for use by this ARC and its future revisions.
-
-A Contract may contain two special methods, `_optIn` and
-`_closeOut`. These methods describe how callers can opt-in to
-(allocate local state), or close-out from (deallocate local state),
-such apps.  They may have any arguments or return values, but there
-must be no more than one method with each name.
-
-In the absence of such methods, ARC-4 Contracts **SHOULD** allow
-accounts to perform the corresponding action with no arguments, or by
-setting the associated OnCompletion value (OptIn = 1, CloseOut = 2)
-while calling a method of the contract. ARC-4 Contracts **MAY** allow
-OnCompletion values to be set to other values in order to support
-methods that delete or update the contract. Of course, great care
-should be taken when allowing these operations.
+(specifically, they must have unique method selectors).
+
+Methods in a Contract that originate from an implemented Interface **MUST**
+contain every allowed OnCompletion action from the Interface's definition
+of the method (though the Contract's program **MAY** reject some of these
+OnCompletion actions at runtime if they do not apply to the Contract's
+implementation). Additionally, the Contract's method definition **MAY**
```

Comment:
I don't understand what this means at all.  They must specify that they allow the operation, but they might not actually allow the operation?